lib/client.ts

```diff
@@ -55,31 +57,53 @@ class Client {
 
     send(requestCommand: string, requestData: any, destinationTwinId: number, expirationMinutes: number) {
         // create new envelope with given data and destination
-        const envelope = new ClientEnvelope(this.source, this.signer, destinationTwinId, requestCommand, requestData, expirationMinutes);
+        const envelope = new Envelope({
+            uid: uuidv4(),
+            timestamp: Math.round(Date.now() / 1000),
+            expiration: expirationMinutes * 60,
+            source: this.source,
+            destination: new Address({ twin: destinationTwinId })
+        });
+        if (requestCommand) {
+            envelope.request = new Request({ command: requestCommand })
+        }
+        if (requestData) {
+            envelope.plain = new Uint8Array(Buffer.from(requestData));
+
+        }
+        const clientEnvelope = new ClientEnvelope(this.signer, envelope, this.chainUrl);
         // send enevelope binary using socket
-        this.con.send(envelope.serializeBinary());
+        this.con.send(clientEnvelope.serializeBinary());
         // add request id to responses map on client object
-        this.responses.set(envelope.uid, envelope)
-        return envelope.uid;
+        this.responses.set(clientEnvelope.uid, clientEnvelope)
+        return clientEnvelope.uid;
 
     }
 
     read(requestID: string) {
         return new Promise(async (resolve, reject) => {
-            let envelope: Envelope = this.responses.get(requestID)
+            let envelope: ClientEnvelope = this.responses.get(requestID)
             // check if envelope in map has a response  
             const now = new Date().getTime();
             while (envelope && new Date().getTime() < now + envelope.expiration * 1000) {
                 envelope = this.responses.get(requestID)
                 if (envelope && envelope.response) {
 
-                    const dataReceived = envelope.plain;
-                    if (dataReceived) {
-                        const decodedData = new TextDecoder('utf8').decode(Buffer.from(dataReceived))
-                        const responseString = JSON.parse(decodedData);
+
+                    const verified = await envelope.verify(envelope.signature)
```

Comment:
why does the verify method take the signature if it's already on the envelope object?